napari/layers/labels/labels.py

```diff
@@ -148,7 +148,9 @@ class Labels(_ImageBase):
         to background.
     color : dict of int to str or array
         Custom label to color mapping. Values must be valid color names or RGBA
-        arrays.
+        arrays. While there is no limit to the number of custom labels, the
+        the layer will render incorrectly if they map to more than 1024 distinct
+        colors.
```

Comment:
Do you think we could/should emit a warning in the function if there are more than 1024 colors ?

Reply:
Sorry, it's un utils, I see it now. Nevermind.

Reply:
There was a question in my mind as to where to generate the error. Given the limit is on the vispy side, I placed in the code generating the control points & colors for vispy.